### PR TITLE
[WIP] adding app gateway route table facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY

Adding app gateway route table facts module.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_appgatewayroutetable_facts

##### ANSIBLE VERSION
2.7

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+class AzureRMRouteTablesFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            route_table_name=dict(
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py:116]: name?

File: lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        self.mgmt_client = None
+        self.resource_group = None
+        self.route_table_name = None
+        self.expand = None
+        super(AzureRMRouteTablesFacts, self).__init__(self.module_arg_spec)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(NetworkManagementClient,
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py:138]: move to common

File: lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if (self.resource_group is not None and
+                self.route_table_name is not None):
+            self.results['route_tables'] = self.get()
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_appgwroutetable_facts.py:143]: else? list_all and list_by_resourcegroup?

**Review by yuwzho — CHANGES_REQUESTED** [review]

**Comment by zikalino** [comment]
@yuwzho not ready for review yet :-)
could you check firewall rules for MySQL and PostgreSQL?

**Pull request closed without merging**